10up/classifai
# Enhancement/566 Confirm Classification before Save

### Description of the Change

* This is an enhancement PR that adds a pop-up when the classify post button is clicked and displays the terms coming from the API along with the existing terms. So users can add or remove terms and when they're expectations meet they can simply save it.
* In the popup, the terms with the prefix "[AI]" stand for the terms that are suggested by the API and do not exist in the post at the moment.
* Only those taxonomies will be displayed that are enabled from the settings.
* If users do not open the classification modal/popup, we remind them on the pre-publish panel as a suggestion to classify the post.
* The popup is also opened once the response is received.
* Users will see a notice when no unique terms are received from AI or all the received terms are already saved.
* Also bumps minimum WP version from 5.8 to 6.1.

### Screenshots

![image](https://github.com/10up/classifai/assets/25176325/4ffcb506-258f-46bf-ad9b-2c6e78cbdc93)

![image](https://github.com/10up/classifai/assets/25176325/6876643b-e9c7-4fcd-98d6-d756581fdeba)

![image](https://github.com/10up/classifai/assets/25176325/ee24eb20-dad2-43ec-9324-c97781c4575f)


Closes #566

### How to test the Change
* Click a "Classify Post" button from the right sidebar in a post with enough content and confirm the popup shows the terms from ClassfAI and also includes the existing terms.
* Try to add/remove the terms and save and see if the final set of terms is applied to the post.
* Reload the page and click the Publish button, you should see the same suggestion dropdowns in the Pre-Publish panel.

### Changelog Entry
> Added - Confirm Classification before Save
> Change - Bump minimum WP version from 5.8 to 6.1

### Credits

Props @jeffpaul @dkotter @faisal-alvi 

### Checklist:
